cmd/clusterctl/client/cluster/crd_migration_test.go

```diff
@@ -111,6 +111,28 @@ func Test_CRDMigrator(t *testing.T) {
 			},
 			wantIsMigrated: false,
 		},
+		{
+			name: "No-op if new CRD adds a new versions and unserves the old",
+			currentCRD: &apiextensionsv1.CustomResourceDefinition{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+				Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+					Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
+						{Name: "v1alpha1", Storage: true, Served: true},
+					},
+				},
+				Status: apiextensionsv1.CustomResourceDefinitionStatus{StoredVersions: []string{"v1alpha1"}},
+			},
+			newCRD: &apiextensionsv1.CustomResourceDefinition{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+				Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+					Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
+						{Name: "v1beta1", Storage: true, Served: true}, // v1beta1 is being added
+						{Name: "v1alpha1", Served: false},              // v1alpha1 still exists but is not served anymore
+					},
+				},
+			},
+			wantIsMigrated: false,
+		},
 		{
 			name: "Fails if new CRD drops current storage version",
```

Comment:
Can we simplify the test cases? There is a lot of misleading language in the test names now, e.g. "Migrate CRs if their storage version is removed from the CRD" (that we migrate is now solely determined by the comparison of the status field and the storage version)